control/celeryapp.py

```diff
@@ -59,6 +66,28 @@ def deliverable_route(veda_id, encode_profile):
     veda_deliver.run()
 
 
+@app.task(name='ingest_worker')
+def ingest(studio_upload_id, video_metadata, course_id, course_hex):
+    """
+    Ingest a video in s3.
+    Arguments:
+        - studio_upload_id: a valid studio upload ID
+        - video_metadata: video metadata
+        - course_id: course id identifying a course run
+        - course_hex: studio_hex identifying course runs
+    """
+    course = get_or_create_course(
```

Comment:
Shouldn't the course be guaranteed created at this moment? Why do we still need to do `get_or_create`?

Reply:
Yes, you're right.

Reply:
Ah, you know why I had a get_or_create here too? Because the get_or_create method is a monster that relies on some complicated interaction between a course hex and a course ID.